adapter/repository/in-memory-simple/src/test/java/org/mobilitydata/gtfsvalidator/db/NoticeMemoryConsumptionTest.java

```diff
@@ -0,0 +1,140 @@
+package org.mobilitydata.gtfsvalidator.db;
+
+import org.junit.jupiter.api.Test;
+import org.mobilitydata.gtfsvalidator.usecase.notice.IntegerFieldValueOutOfRangeNotice;
+import org.mobilitydata.gtfsvalidator.usecase.notice.base.ErrorNotice;
+import org.mobilitydata.gtfsvalidator.usecase.notice.base.InfoNotice;
+import org.mobilitydata.gtfsvalidator.usecase.notice.base.Notice;
+import org.mobilitydata.gtfsvalidator.usecase.notice.base.WarningNotice;
+import org.mobilitydata.gtfsvalidator.usecase.port.ValidationResultRepository;
+
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.List;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+public class NoticeMemoryConsumptionTest {
+
+    private void generateNotices(MockValidationResultRepo mockValidationResultRepo, int numberOfNotices) {
+        for (int i = 0; i < numberOfNotices; i++) {
+            mockValidationResultRepo.addNotice(new IntegerFieldValueOutOfRangeNotice("filename",
+                    "fieldname",
+                    "entity_id",
+                    0,
+                    100,
+                    101)
+            );
+        }
+    }
+
+    @Test
+    public void creationOf100NoticeShouldNotRaiseException() {
+
+        MockValidationResultRepo mockValidationResultRepo = new MockValidationResultRepo();
+
+        generateNotices(mockValidationResultRepo, 100);
+        mockValidationResultRepo.getAll();
+    }
+
+    @Test
+    public void creationOf1000NoticeShouldNotRaiseException() {
+
+        MockValidationResultRepo mockValidationResultRepo = new MockValidationResultRepo();
+
+        generateNotices(mockValidationResultRepo, 1_000);
+        mockValidationResultRepo.getAll();
+
+    }
+
+    @Test
+    public void creationOf10000NoticeShouldNotRaiseException() {
+
+        MockValidationResultRepo mockValidationResultRepo = new MockValidationResultRepo();
+
+        generateNotices(mockValidationResultRepo, 10_000);
+        mockValidationResultRepo.getAll();
+
+    }
+
+    @Test
+    public void creationOf100000NoticeShouldNotRaiseException() {
+
+        MockValidationResultRepo mockValidationResultRepo = new MockValidationResultRepo();
+
+        generateNotices(mockValidationResultRepo, 100_000);
+        mockValidationResultRepo.getAll();
+
+    }
+
+    @Test
+    public void creationOf1_000_000NoticeShouldNotRaiseException() {
+
+        MockValidationResultRepo mockValidationResultRepo = new MockValidationResultRepo();
+
+        generateNotices(mockValidationResultRepo, 1_000_000);
+        mockValidationResultRepo.getAll();
+    }
+
+    @Test
```

Comment:
I'm not entirely sure I understand what this test is trying to do. Right now it just proves that we can't handle a certain number of records, but that's going to be very machine-dependent - it may pass on some machines and fail on others.

It seems like the following alternate versions would be more useful:
1. Benchmark and print memory consumption of the test - Maybe using something like `Runtime.getRuntime().totalMemory() - Runtime.getRuntime().freeMemory()` (see https://stackoverflow.com/questions/17374743/how-can-i-get-the-memory-that-my-java-program-uses-via-javas-runtime-api)? This would give us a machine-independent history of how much memory we're using
1. Fail gracefully - Have a test that ensures if we do run out of memory when validating a feed that we're able to fail gracefully and still provide results. This would need to be mocked to force the error to ensure that the test passes on all machines.

Reply:
We will implement 1. right away, and shall open an issue for 2.

Reply:
#77